src/script/components/TextInput.tsx

```diff
@@ -0,0 +1,194 @@
+            color: 'var(--text-input-alert)',
+            left: 0,
+            lineHeight: '14px',
+            position: 'absolute',
+            textTransform: 'unset',
+          }}
+        >
+          {errorMessage}
+        </span>
+      )}
+      <input
+        className="text-input"
+        css={{
+          '&[disabled]': {},
+          '&::placeholder': {
+            color: 'var(--text-input-placeholder)',
+          },
+          '&:hover': {
+            borderColor: !disabled && 'var(--text-input-border-hover)',
+          },
+          '&:focus, &:active': {
+            '& + label': {
+              color: !disabled && 'var(--blue-500)',
+            },
+            borderColor: !disabled && 'var(--blue-500)',
+          },
+          '::placeholder': {
+            // Chrome, Firefox, Opera, Safari 10.1+
+            color: 'var(--text-input-placeholder)',
+            opacity: 1, // Firefox
+          },
+          ':-ms-input-placeholder': {
+            // Internet Explorer 10-11
+            color: 'var(--text-input-placeholder)',
+          },
+          '::-ms-input-placeholder': {
+            // Microsoft Edge
+            color: 'var(--text-input-placeholder)',
+          },
+          background: disabled ? 'var(--text-input-disabled)' : 'var(--text-input-background)',
+          border: '1px solid',
+          borderColor: changedColor || 'var(--text-input-border)',
+          borderRadius: 12,
+          color: 'var(--text-input-color)',
+          outline: 'none',
+          padding: '12px 16px',
+          width: '100%',
+        }}
+        disabled={disabled}
+        id={name}
+        name={name}
+        value={value}
+        onChange={onChange}
+        placeholder={placeholder}
+      />
+      <label
+        className="label-medium"
+        css={{
+          color: changedColor || 'var(--text-input-label)',
+          display: 'flex',
+          flexDirection: 'column',
+          marginBottom: 2,
+        }}
+        htmlFor={name}
+      >
+        {label}
+      </label>
+      {isFilled && !isSuccess && !isError && (
+        <button
+          css={{
+            alignItems: 'center',
+            background: 'var(--text-input-color)',
+            border: 'none',
+            borderRadius: '50%',
+            bottom: 42,
+            display: 'flex',
+            height: 16,
+            justifyContent: 'center',
+            margin: 0,
+            padding: 0,
+            position: 'absolute',
+            right: 16,
+            width: 16,
+          }}
+          onClick={onCancel}
+        >
+          <Icon.Close css={{fill: 'var(--text-input-background)', height: 8, width: 8}} />
+        </button>
+      )}
+      {isSuccess && !isError && <CheckIcon css={iconCss} color={COLOR.TEXT} />}
+      {isError && <Icon.ExclamationMark css={iconCss} color={COLOR.TEXT} />}
+    </div>
+  );
+};
+
+export default TextInput;
+
+registerReactComponent<UserInputProps>('text-input', {
```

Comment:
I don't see where this component is used :thinking:

Reply:
I don't use it in this PR - I'll use it in the preferences page once it's merged

Reply:
:ok: